scripts/o3de/o3de/register.py

```diff
@@ -683,6 +697,7 @@ def register(engine_path: pathlib.Path = None,
              default_third_party_folder: pathlib.Path = None,
              external_subdir_engine_path: pathlib.Path = None,
              external_subdir_project_path: pathlib.Path = None,
+             external_subdir_gem_path: pathlib.Path = None,
```

Comment:
This change will break several functions in Project Manager's `PythonBindings.cpp` that call `register` directly expecting a particular parameter order.  Search for "register" in the file and you'll find a few (`AddGemRepo`, `RemoveGemRepo`,`SetEngineInfo`,`GemRegistration`,`AddProject`,`RemoveProject`).  

We need to update all the functions in there that call "register" to use `using namespace pybind11::literals;` so that we can just reference the function parameters that are used instead of relying on  particular order which will break every time the parameter order is changed.  See the `CreateProject` function for an example of how to use literals - you basically add the namespace and then indicate the parameter name in quotes with a `_a` after it.

Reply:
Yeah we should update these in PythonBindings.cpp blocks to used named parameters
* https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L561-L579
* https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L705-L722
* https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L773-L791
* https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Source/PythonBindings.cpp#L1395-L1413